datafusion/core/tests/physical_optimizer/filter_pushdown/mod.rs

```diff
@@ -734,6 +734,366 @@ async fn test_topk_dynamic_filter_pushdown() {
     );
 }
 
+#[tokio::test]
+async fn test_hashjoin_dynamic_filter_pushdown() {
+    use datafusion_common::JoinType;
+    use datafusion_physical_plan::joins::{HashJoinExec, PartitionMode};
+
+    // Create build side with limited values
+    let build_batches = vec![record_batch!(
+        ("a", Utf8, ["aa", "ab"]),
+        ("b", Utf8, ["ba", "bb"]),
```

Comment:
We may add some Utf8View  fields testing cases, because our default mapping already changing to Utf8View.